File: corporate-reporting/src/types/Dependency.ts
import { IReportingEntityWithDeliveryDates, ReportingEntityWithDeliveryDates } from "./ReportingEntity";
import { IProject } from "./Project";
import { IUser } from "./User";
import { IDependencyUpdate } from "./DependencyUpdate";

export interface IDependency extends IReportingEntityWithDeliveryDates {
    ProjectID: number;
    ThirdParty: string;
    RagOptionID: number;
    LeadUserID: number;
    StartDate: Date;
    EndDate: Date;

    Project?: IProject;
    LeadUser?: IUser;
    DependencyUpdates?: IDependencyUpdate[];
}

export class Dependency extends ReportingEntityWithDeliveryDates implements IDependency {
    public ProjectID = null;
    public ThirdParty = '';
    public RagOptionID = null;
    public LeadUserID = null;
    public StartDate = null;
    public EndDate = null;
    public Contributors = [];
}

File: ORB.Core/Models/Dependency.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public class Dependency : ReportingSubEntity
    {
        public Dependency()
        {
            DependencyUpdates = new HashSet<DependencyUpdate>();
        }

        public int ProjectID { get; set; }
        public string ThirdParty { get; set; }
        public int? RagOptionID { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public DateTime? BaselineDate { get; set; }
        public DateTime? ForecastDate { get; set; }
        public DateTime? ActualDate { get; set; }

        public virtual Project Project { get; set; }
        public virtual RagOption RagOption { get; set; }
        public virtual ICollection<DependencyUpdate> DependencyUpdates { get; set; }
    }
}
